src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private Node<T> currentNode;
+    private Node<T> previousNode;
+    private Node<T> nextNode;
```

Comment:
The field 'nextNode' seems to be redundant. Consider removing it if it's not necessary for your implementation.